packages/kbn-alerts-ui-shared/src/rule_form/rule_page/rule_page.test.tsx

```diff
@@ -125,4 +125,33 @@ describe('rulePage', () => {
     fireEvent.click(screen.getByTestId('rulePageReturnButton'));
     expect(onCancel).toHaveBeenCalled();
   });
+
+  test('should display discard changes modal only if changes are made in the form', async () => {
```

Comment:
Could you please add another for the opposite scenario? No changes are made and the modal is not shown.

Reply:
Done, test added. Thanks!